src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -1737,3 +1737,26 @@ case class If(pos: Position, cond: AST, thenTree: AST, elseTree: AST)
     }
   }
 }
+
+case class Splat(pos: Position, lhs: AST) extends AST(pos, lhs) {
+  override def typecheckThis(): BaseType = {
+    lhs.`type` match {
+      case t: TStruct => TSplat(t)
+      case t => parseError(
+        s"""splatting ( <identifier>.* ) operations are only supported on `Struct'
+            |  Found `$t'
+         """.stripMargin)
+    }
+  }
+
+  override def eval(ec: EvalContext): () => Any = {
+    val nElem = `type`.asInstanceOf[TSplat].struct.size
```

Comment:
I'm happy to rework it, but do let me know what you want soon.  This is integral to fixing the tutorial.